amazon-kinesis-client/src/main/java/software/amazon/kinesis/lifecycle/ShardConsumer.java

```diff
@@ -392,6 +397,7 @@ private synchronized void executeTask(ProcessRecordsInput input) {
             }
             taskOutcome = resultToOutcome(result);
         }
+        taskExecutionListener.onTaskEnd(currentState, shardInfo);
```

Comment:
- Documentation on ConsumerTask.java says that any exception during task.call() will be captured in TaskResult, returned by task.call(). I also looked at all *.Task classes to confirm that, so it would be safe to assume that it won't throw exception and onTaskBegin will always be accompanied by onTaskEnd in all known cases.

- onTaskBegin and onTaskEnd should always be called in the same order.